SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,252 @@
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    simd<DataT, NumElems> simd_by_init =
+        simd<DataT, NumElems>(ref_data, alignment);
+    simd_by_init.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in variable declaration
+// context.
+struct var_decl {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    simd<DataT, NumElems> simd_by_var_decl(ref_data, alignment);
+    simd_by_var_decl.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in rvalue in an
+// expression context.
+struct rval_in_expr {
+  static std::string get_description() { return "rvalue in an expression"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    simd<DataT, NumElems> simd_by_rval;
+    simd_by_rval = simd<DataT, NumElems>(ref_data, alignment);
+    simd_by_rval.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in const reference
+// context.
+class const_ref {
+public:
+  static std::string get_description() { return "const reference"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    call_simd_by_const_ref<DataT, NumElems>(
+        simd<DataT, NumElems>(ref_data, alignment), out);
+  }
+
+private:
+  template <typename DataT, int NumElems>
+  static void
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T, typename Alignment>
+struct Kernel_for_load_ctor;
+
+namespace alignment {
+
+struct element {
+  template <typename DataT, int NumElems> static size_t get_size() {
+    return alignof(DataT);
+  }
+  template <typename DataT> static constexpr auto get_value() {
+    return element_aligned;
+  }
+};
+
+struct vector {
+  template <typename DataT, int NumElems> static size_t get_size() {
+    return alignof(simd<DataT, NumElems>);
+  }
+  template <typename DataT> static constexpr auto get_value() {
+    return vector_aligned;
+  }
+};
+
+struct overal {
+  template <typename DataT, int NumElems> static size_t get_size() {
+    return alignof(std::max_align_t);
+  }
```

Comment:
```suggestion
  template <typename, int> static size_t get_size() {
    return alignof(std::max_align_t);
  }
```

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/769/commits/d6ef82c019b58ae681f2e60ed94fc8eb862dc2ef#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7R113)